File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

var dnaToRna = map[rune]rune{'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}

func ToRNA(dna string) string {
	dnaStrand := []rune(dna)
    rnaStrand := make([]rune, len(dnaStrand))
    for i, dnaNucleotide := range dnaStrand {
        rnaStrand[i] = dnaToRna[dnaNucleotide]
    }
    return string(rnaStrand)
}

File: solutions/lua/rna-transcription/1/rna-transcription.lua
local DNA_TO_RNA = {
    G = 'C',
    C = 'G',
    T = 'A',
    A = 'U'
}

return function(dna_strand)

    local rna_strand = {}
    for i, dna in ipairs { string.byte(dna_strand, 1, -1) } do
        rna_strand[i] = DNA_TO_RNA[string.char(dna)]
    end

    return table.concat(rna_strand, '')
end
